plugin/metrics/prometheus/factory.go

```diff
@@ -66,3 +67,29 @@ func (f *Factory) Initialize(logger *zap.Logger) error {
 func (f *Factory) CreateMetricsReader() (metricsstore.Reader, error) {
 	return prometheusstore.NewMetricsReader(f.options.Configuration, f.logger, f.tracer)
 }
+
+func (f *Factory) configureFromOptions(o *Options) {
+	f.options = o
+}
+
+func NewFactoryWithConfig(
```

Comment:
nit: if we introduce this function, I would prefer if the other constructor would delegate to this one if possible, to minimize divergence between v1 and v2 code paths.

Reply:
@yurishkuro I am not sure if this is possible as `NewFactoryWithConfig` requires ` cfg config.Configuration, logger *zap.Logger ` but we don't have access to those in `NewFactory`

Reply:
ok